contractsV5/KyberNetwork.sol

```diff
@@ -523,28 +524,107 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
+        uint percentageSoFar; //need this to calculate feePayingPercentage
+        uint splitAmountAfterFee; //to calculate rate with fee
+        uint splitAmountNoFee; //to calculate rate without fee
+        uint destAmountNoFee;
+        tradeData.tokenToEth.decimals = getDecimals(src);
+        tradeData.ethToToken.decimals = getDecimals(dest);
 
         // token to Eth
         ///////////////
         // if hinted reserves, find rates.
-        // calculate accumulated amounts
+        if (tradeData.tokenToEth.addresses.length > 0) {
+            for (i = 0; i < tradeData.tokenToEth.addresses.length; i++) {
+                reserve = tradeData.tokenToEth.addresses[i];
+                //calculate split and corresponding trade amounts
+                splitAmountNoFee = (i == tradeData.tokenToEth.splitValuesPercent.length) ? (srcAmount - amountSoFarNoFee) : tradeData.tokenToEth.splitValuesPercent[i] * srcAmount /  100;
+                amountSoFarNoFee += splitAmountNoFee;
+                tradeData.tokenToEth.rates[i] = reserve.getConversionRate(src, dest, splitAmountNoFee, block.number);
+                tradeData.tradeWeiAmount += calcDestAmountWithDecimals(tradeData.tokenToEth.decimals, ETH_DECIMALS, splitAmountNoFee, rate);
+
+                //account for fees
+                if (tradeData.tokenToEth.isPayingFees[i]) {
+                    tradeData.feePayingPercentage += (i == tradeData.tokenToEth.splitValuesPercent.length) ? (100 - percentageSoFar) : tradeData.tokenToEth.splitValuesPercent[i];
+                }
+                percentageSoFar += tradeData.tokenToEth.splitValuesPercent[i];
+            }
+        } else {
+            // else find best rate
+            (reserve, rate, isPayingFees) = searchBestRate(src, ETH_TOKEN_ADDRESS, srcAmount, tradeData.usePermissionless);
+            // save into tradeData
+            tradeData.tokenToEth.addresses[0] = reserve;
+            tradeData.tokenToEth.rates[0] = rate;
+            tradeData.tokenToEth.splitValuesPercent[0] = 100; //max percentage amount
+            tradeData.tradeWeiAmount = calcDestAmountWithDecimals(tradeData.tokenToEth.decimals, ETH_DECIMALS, srcAmount, rate);
+
+            //account for fees
+            if (isPayingFees) {
+                tradeData.feePayingPercentage = 100; //max percentage amount for token -> ETH
+            }
+        }
 
-        // else find best rate
+        //add percentage for ETH -> token reserves
+        //if there is no ETH -> token reserve specified, percentage added will be zero, to be handled after searching best rate
+        for (i = 0; i < tradeData.ethToToken.addresses.length; i++) {
+            if (tradeData.ethToToken.isPayingFees[i]) {
+                tradeData.feePayingPercentage += (i == tradeData.ethToToken.splitValuesPercent.length-1) ? (100 - percentageSoFar) : tradeData.ethToToken.splitValuesPercent[i];
+            }
+            percentageSoFar += tradeData.tokenToEth.splitValuesPercent[i];
```

Comment:
why do we need the local one?
can't save all in tradeData.feePayingPercentage?

Reply:
Hmmmm you're right, it's not needed.

Reply:
No it can't. Because we use N-1 instead of N for splitValuesPercent, I don't know the trade percentage of the last reserve without the variable. If I just store it in tradeData, then the feePayingPercentage will **always** be 100% if the last reserve is fee paying.

Example: 1st reserve is feeless (60% of tradeQty), 2nd reserve is fee paying (40% of tradeQty). 
Expected: feePayingPercentage = 40%
Actual:  feePayingPercentage = 100%

Reply:
This can be solved if we store splitValuesPercent to be N... I think.